troposphere/static/js/components/modals/instance/launch/components/ResourcesForm.jsx

```diff
@@ -135,13 +135,11 @@ const ResourcesForm = React.createClass({
         </label>
         );
     },
-    render: function() {
-        let { identity, identityList, onIdentityChange, providerSize, providerSizeList, onSizeChange, } = this.props;
-        return (
-        <form>
-            {globals.USE_ALLOCATION_SOURCES
-             ? this.renderAllocationSourceMenu()
-             : null}
+    renderIdentity() {
```

Comment:
Aesthetic complaint:
Conceptually `renderIdentity` either renders a component or renders nothing, and `renderIdentity` should always render a component. Move the boolean flag outside the call to renderIdentity.

In the jsx I prefer something like:
```
{ hasProjectSharing() ? this.renderIdentity() : null }
```

This make it clear that you're not always rendering a component.